state/validatorsInfoMap.go

```diff
@@ -0,0 +1,224 @@
+package state
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+)
+
+type shardValidatorsInfoMap struct {
+	mutex      sync.RWMutex
+	valInfoMap map[uint32][]ValidatorInfoHandler
+}
+
+// NewShardValidatorsInfoMap creates an instance of shardValidatorsInfoMap which manages a
+// <shardID, validatorsInfo> map internally
+func NewShardValidatorsInfoMap(numOfShards uint32) *shardValidatorsInfoMap {
+	return &shardValidatorsInfoMap{
+		mutex:      sync.RWMutex{},
+		valInfoMap: make(map[uint32][]ValidatorInfoHandler, numOfShards),
+	}
+}
+
+// TODO: Delete this once map[uint32][]*ValidatorInfo is completely replaced with new interface
+
+// CreateShardValidatorsMap creates an instance of shardValidatorsInfoMap which manages a shard validator
+// info map internally.
+func CreateShardValidatorsMap(input map[uint32][]*ValidatorInfo) *shardValidatorsInfoMap {
+	ret := &shardValidatorsInfoMap{valInfoMap: make(map[uint32][]ValidatorInfoHandler, len(input))}
```

Comment:
also here, we do not need the call to len(input)

Reply:
I think here it is ok to call len to allocate memory for the map, since they will have the same size, right?